streamer/pulp/streamer/server.py

```diff
@@ -225,7 +226,23 @@ def close(self):
         """
         Forward the call to close the 'file' to the request.finish method.
         """
-        reactor.callFromThread(self.request.finish)
+        reactor.callFromThread(self.finish_wrapper)
+
+    def finish_wrapper(self):
```

Comment:
You may be interested to know about this, even though we obviously can't use it: https://docs.python.org/3.4/library/contextlib.html#contextlib.suppress